src/Type/Definition/StringType.php

```diff
@@ -17,45 +17,29 @@
 
 class StringType extends ScalarType
 {
-    /** @var string */
     public $name = Type::STRING;
 
-    /** @var string */
     public $description =
         'The `String` scalar type represents textual data, represented as UTF-8
 character sequences. The String type is most often used by GraphQL to
 represent free-form human-readable text.';
 
-    /**
-     * @param mixed $value
-     *
-     * @return mixed|string
-     *
-     * @throws Error
-     */
-    public function serialize($value)
+    public function serialize($value): string
     {
         $canCast = is_scalar($value)
             || (is_object($value) && method_exists($value, '__toString'))
             || $value === null;
 
         if (! $canCast) {
-            throw new Error(
+            throw new SerializationError(
                 'String cannot represent value: ' . Utils::printSafe($value)
             );
         }
 
         return (string) $value;
     }
 
-    /**
-     * @param mixed $value
```

Comment:
I'd keep those mixed so they can be converted to `(mixed $value)` on php 8 automatically.

With this change we wouldn't be able to distinguish intentional mixed versus forgotten type. Also, consumers can `@inheritDoc`

Reply:
The parent method still has the explicit `mixed` annotation.

Reply:
But the thing with php is to specify types even in implementations but that's related to previous discussion on disabling missing type checks.  
We can have @inheritDoc here so it explicitly says we want `mixed` here. 

With this change it's ambiguous: is it forgotten or inherited? What if parent type narrows, should it be narrowed here as well?

Reply:
Good tooling I use assumes the parent PHPDoc is inherited, unless specifically overwritten. Perhaps we can document and reify that assumption as part of the contribution guidelines?

Reply:
Do as you think is the best but personally I don't like trading automated checks for a piece of text few people read 🤷🏾‍♀️

Reply:
I am fairly confident we will keep this project moving towards stronger types without duplicating code.